### PR TITLE
Adding identitytoolkit v1 and v2 APIs

### LINKED ISSUE #543: Adding identitytoolkit v1 and v2 APIs
Adding these APIs would make this package useful for server-side dart and user management with firebase-auth.

- identitytoolkit:v1
   https://cloud.google.com/identity-platform/docs/apis/
- identitytoolkit:v2
  https://cloud.google.com/identity-platform/docs/apis/

### PR DESCRIPTION
**Pull request by slightfoot**
Resolves #543

### CODE REVIEW

File: discoveryapis_generator/lib/src/dart_resources.dart [hunk 1]
```diff
@@ -670,7 +670,7 @@ DartResourceMethod _parseMethod(
 
   // Check if we have a request object, if so parse it's type.
   MethodParameter? dartRequestParameter;
-  if (method.request != null) {
```
Comment (kevmoo) [inline discoveryapis_generator/lib/src/dart_resources.dart:673]: Can you explain the diff in the generator?

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,7 @@ pubspec.lock
 test_integration/*.json
 test_integration/client_id*.yaml
 test_integration/config.yaml
+
+# IDE files
```
Comment (kevmoo) [inline .gitignore:10]: please, no!

These should be in your own global ignore file

Comment (slightfoot) [reply .gitignore]: Removed and thanks, many years using git and didn't know you could have a global ignore file!

**Comment by kevmoo** [comment]
Please rebase on HEAD!

File: discoveryapis_generator/lib/src/dart_resources.dart (hunk 1 above)
Comment (slightfoot) [reply discoveryapis_generator/lib/src/dart_resources.dart:673]: I was follow the contrib guide and ran the generator but it crashed out with v1/v2 json files in play. I decided to debug it expecting I'd nee to contact you to create some diff/patches. But debugging and stepping it I found the code would pass in nullable references but always treat them as non-null. 

I've also update those type declarations to make this clear.

**Comment by slightfoot** [comment]
@kevmoo rebased on HEAD.

File: discoveryapis_generator/lib/src/generated_googleapis/discovery/v1.dart [hunk 3]
```diff
@@ -1105,10 +1105,10 @@ class RestMethodMediaUpload {
 /// The schema for the request.
 class RestMethodRequest {
   /// Schema ID for the request schema.
-  core.String? P_ref;
```
Comment (kevmoo) [inline discoveryapis_generator/lib/src/generated_googleapis/discovery/v1.dart:1108]: Careful! these are generated!

Comment (slightfoot) [reply discoveryapis_generator/lib/src/generated_googleapis/discovery/v1.dart:1108]: Oh, I'll change them back

Comment (slightfoot) [reply discoveryapis_generator/lib/src/generated_googleapis/discovery/v1.dart]: done

**Pull request merged**